### PR TITLE
Game menu slide

### PR DESCRIPTION
**Pull request by sarahgos**
Implemented the sliding functionality of the menu. I still have not figured out how to display it underneath the overlay.

### CODE REVIEW

**Review by studioant — APPROVED** [review]
This is cool Sarah! Great contribution, well done

File: menu.h [hunk 1]
```diff
@@ -200,6 +242,86 @@ class Menu {
             draw_overlay(button->config);
     }
 
+        // Method to update the sprite positions and draw sprite.
+    void update_slide(sprite button_sprite, int position)
+    {
+        // Show the base layer of sprite.
```
Comment (studioant) [inline menu.h:248]: To solve the overlay issue, we could just turn the overlay off when sliding. eg:
`this->_overlayActive = false;`

Comment (sarahgos) [reply menu.h:249]: Thanks Anthony, I have now added this.

**Comment by Delcari** [comment]
Seems to be an issue which occurs when you try to move left-right in quick succession. 
Graphical Issue with sliding animation

**Comment by sarahgos** [comment]
@Delcari Well this certainly isn't ideal. I will have a look over it tomorrow and will probably add some booleans to prevent constant arrow input. Not sure about the segmentation fault, I did experience it once and wasn't sure if it was a random one, but obviously not. I will click through it tomorrow and find out where that is occurring. Thanks for the testing Riley!

**Comment by sarahgos** [comment]
@Delcari I believe I have fixed these issues, when you get a chance could you please review? Thanks.

**Pull request merged**